src/Components/Instagram-authentication.js

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { auth } from "../firebase";
+
+export default class InstagramAuth extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      emailInputValue: "",
+      passwordInputValue: "",
+      newUser: true,
+      errorCode: "",
+      errorMessage: "",
+    };
+  }
+
+  handleInputChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    const closeAuthentication = () => {
+      this.setState({
+        emailInputValue: "",
+        passwordInputValue: "",
+        newUser: true,
+        errorCode: "",
+        errorMessage: "",
+      });
```

Comment:
As mentioned above, use a defaultState object instead with these values
```suggestion
      this.setState(defaultState);
```